tests/base_scraper/test_base_scraper.py

```diff
@@ -113,5 +113,35 @@ def test_optional_authentication(self):
         self.assertTrue(os.path.isfile(os.path.join(self.temp_dir,
                                                     'ci.mozilla.org')))
 
+    def test_destination(self):
+        """Test for various destination scenarios"""
+
+        filename = 'download_test.txt'
+        test_url = urljoin(self.wdir, filename)
+
+        # destination is directory
+        scraper = mozdownload.DirectScraper(url=test_url,
+                                            destination=self.temp_dir,
+                                            version=None,
+                                            log_level='ERROR')
+        self.assertEqual(scraper.target, os.path.join(self.temp_dir, filename))
+
+        # destination is file
+        destination = os.path.join(self.temp_dir, filename)
+        scraper = mozdownload.DirectScraper(url=test_url,
+                                            destination=destination,
+                                            version=None,
+                                            log_level='ERROR')
+        self.assertEqual(scraper.target, destination)
+
+        # destination directory does not exist
```

Comment:
I would also like to see the case when a filename has been specified and the folder does not exist yet. Maybe even multiple levels deep, so we can ensure that we correctly create a folder tree.